.github/scripts/utils/string_manipulation.py

```diff
@@ -221,3 +175,56 @@ def replace_after_string_in_url(url: str, new_str: str, previous_str: str = "ega
             new_parts.append(part)
     
     return "/".join(new_parts)
+
+
+def validate_json_file_path(file_path: str) -> None:
+    """
+    Validates if a file exists and is a JSON file
+    """
+    if not os.path.isfile(file_path):
+        raise FileNotFoundError(f"File not found: {file_path}")
+    file_extension = os.path.splitext(file_path)[1].lower()
+    if not file_extension == ".json":
+        raise ValueError(f"File is not a JSON file: {file_path}")
+    
+
+def compare_semantic_versions(old_version: str, new_version: str) -> Union[bool, str]:
```

Comment:
could add unit tests for this string_manipulation.py module to test the functionality of each util method.

Reply:
or actually, you might be able to utilise this module https://github.com/python-semver/python-semver it seems to be well-maintained, but if you think it's not applicable, i'm fine with keeping the custom implementation

Reply:
Added semver